### ISSUE TITLE
ISO8601 timestamps

### ISSUE THREAD

**Issue by ptman** [zfsnap/zfsnap#70]
Would it be possible to use ISO8601 standard timestamps instead of another different format.

**Comment by graudeejs** [comment]
Yes, however that would require quite a lot of modifications especially to ensure both formats are supported (backwards compatibility is very important in this case).

Do you have a good reason for wanting timestamps in ISO8601 format?

**Comment by Alexqw** [comment]
@ptman as @graudeejs said, it would take a decent number of modifications to make that work.

The biggest advantage I can see, is storing everything at UTC0, so cross-site backup schemes that span timezones can work. Is this why you're asking, or would you really prefer this other time format?

The biggest concern would be making sure the math always knew which mode it was in (evaluate this snapshot as UTC mode, this other one in locale mode), since the pools will likely be mixed with older non-UTC and current UTC snapshots.

---Alex

**Comment by ptman** [comment]
Aren't you breaking backwards compat with 2.0 any way? I just prefer using standard when such are available. If one needs to parse the dates using some other util there are standard libraries for parsing ISO8601 without having to write regexes or strptime -strings. ISO8601 is used in many places, so it is easy for me to parse visually. With custom formats I always have to stop and think how this particular program prefers to format dates.

**Comment by graudeejs** [comment]
@ptman, yes, but only how you invoke zfsnap.
New version will work with old snapshots just fine.

However to work around backwards comparability issue - migration script could be written that would rename all zfsnap snapshots to new format.
